### ISSUE TITLE
NullPointerException after "cleaning the network" step?

### ISSUE THREAD

**Issue by cmdcolin** [combogenomics/medusa#20]
I was getting an error when I was trying to in a small set of contigs and compare it to another genome.

The full run log looked like this

```
$ java -Xmx11g -jar medusa.jar -i contigs.fa -f ~/genome/ -o my_new_scaffolds.fa -v
INPUT FILE:contigs.fa
------------------------------------------------------------------------------------------------------------------------
Running MUMmer...1: PREPARING DATA
2,3: RUNNING mummer AND CREATING CLUSTERS
# reading input file "contigs_GCF_001624265.ntref" of length 2888899
# construct suffix tree for sequence of length 2888899
# (maximum reference length is 2305843009213693948)
# (maximum query length is 18446744073709551615)
# process 28888 characters per dot
#....................................................................................................
# CONSTRUCTIONTIME /opt/software/MUMmer/3.23--GCC-4.4.5/mummer contigs_GCF_001624265.ntref 0.84
# reading input file "/mnt/home/me/genome/GCF_001624265.1_ASM162426v1_genomic.fa" of length 777364094
# matching query-file "/mnt/home/me/genome/GCF_001624265.1_ASM162426v1_genomic.fa"
# against subject-file "contigs_GCF_001624265.ntref"
# COMPLETETIME /opt/software/MUMmer/3.23--GCC-4.4.5/mummer contigs_GCF_001624265.ntref 532.68
# SPACE /opt/software/MUMmer/3.23--GCC-4.4.5/mummer contigs_GCF_001624265.ntref 754.05
4: FINISHING DATA
done.
------------------------------------------------------------------------------------------------------------------------
Building the network...done.
------------------------------------------------------------------------------------------------------------------------
Cleaning the network...done.
------------------------------------------------------------------------------------------------------------------------
Exception in thread "main" java.lang.NullPointerException
        at graphs.MyGraph.reverseComplement(Unknown Source)
        at graphs.MyGraph.scaffoldStringSeq(Unknown Source)
        at graphs.MyGraph.readScaffoldsSeq(Unknown Source)
        at Scaffolder.scaffolder(Unknown Source)
        at Scaffolder.runOnTerminal(Unknown Source)
        at Scaffolder.<init>(Unknown Source)
        at Scaffolder.main(Unknown Source)
```

Using python 2.7.8, biopython 1.69, java 8, mummer 3.23

If there's any other info that can help let me know. I didn't see any line numbers in the stack trace so I can't see anything further myself

**Comment by EBosi** [comment]
Hi Colin,
sorry for the (really) late reply. Would you mind sharing your dataset so I
can try a bit of debugging? I have no idea on what could be causing this
issue...
Thank you and sorry again,
Emanuele

**Comment by cmdcolin** [comment]
Hi, I'll send you an email with some data/details!

**Comment by cmdcolin** [comment]
I have been re-running on a different dataset for different purposes than the issue that I opened this one up for. I'll simply close issue for now unless I run into this again

**Comment by jvollme** [comment]
Since I ran into this issue also, and could not find a solution in any of the discussions, but think i might have found the problem, I would like to add my take here:

It seems that medusa has a length-restriction for the names of each fasta-sequence?
I also kept getting this error, when i tried to scaffold a multifasta which included contigs with names longer than 20 characters.
Comparing this consistently failing run with previous runs that worked in the past, the only difference i could find were the lenghts of the sequence names.

When I changed the sequence names to be shorter than 20 characters, everything seemed to work again just fine!
